micrometer-core/src/main/java/io/micrometer/core/instrument/binder/neo4j/Neo4jMetrics.java

```diff
@@ -0,0 +1,118 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.neo4j;
+
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import io.micrometer.core.lang.NonNullApi;
+import io.micrometer.core.lang.NonNullFields;
+
+import java.util.function.Consumer;
+
+import org.neo4j.driver.ConnectionPoolMetrics;
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.Metrics;
+
+/**
+ * {@link MeterBinder} that binds all available Neo4j driver metrics.
+ *
+ * @author Michael J. Simons
+ * @author Gerrit Meier
+ */
+@NonNullApi
+@NonNullFields
+public class Neo4jMetrics implements MeterBinder {
+
+    /**
+     * Prefixed used for driver metrics.
+     */
+    public static final String PREFIX = "neo4j.driver.connections";
+
+    private static final String BASE_UNIT_CONNECTIONS = "connections";
+
+    private final Driver driver;
+
+    private final Iterable<Tag> tags;
+
+    public Neo4jMetrics(String name, Driver driver, Iterable<Tag> tags) {
+        this.driver = driver;
+        this.tags = Tags.concat(tags, "name", name);
+    }
+
+    @Override
+    public void bindTo(MeterRegistry meterRegistry) {
+        Metrics metrics = this.driver.metrics();
+        metrics.connectionPoolMetrics().forEach(this.getPoolMetricsBinder(meterRegistry));
+    }
+
+    Consumer<ConnectionPoolMetrics> getPoolMetricsBinder(MeterRegistry meterRegistry) {
+        return (poolMetrics) -> {
+            Iterable<Tag> poolTags = Tags.concat(this.tags, "poolId", poolMetrics.id());
+
+            // acquisition attempts
+            FunctionCounter.builder(PREFIX + ".acquisition", poolMetrics, ConnectionPoolMetrics::acquired)
+                    .tags(Tags.concat(poolTags, "result", "successful"))
+                    .baseUnit(BASE_UNIT_CONNECTIONS).description("The amount of connections that have been acquired.")
+                    .register(meterRegistry);
+
+            FunctionCounter
+                    .builder(PREFIX + ".acquisition", poolMetrics, ConnectionPoolMetrics::timedOutToAcquire)
+                    .tags(Tags.concat(poolTags, "result", "timedOutToAcquire"))
+                    .baseUnit(BASE_UNIT_CONNECTIONS)
+                    .description("The amount of failures to acquire a connection from a pool within maximum connection "
+                                    + "acquisition timeout.")
+                    .register(meterRegistry);
+
+            // connections successfully created and closed
+            FunctionCounter.builder(PREFIX, poolMetrics, ConnectionPoolMetrics::created)
```

Comment:
Does using "PREFIX" only break elastic? I think so... We'd need a unique suffix to this. I wonder as well whether it makes sense to combine created and closed. I can see it both ways...

Reply:
Hi Jon. I'm chiming in for Gerrit.

Tbh, I don't understand your question. What's the thing about Elastic? 

If we need a third vote, than I'd go for the duplication so that we can see both metrics related to each other.